appium-examples/src/test/java/com/realdevice/image_injection/ImageInjectionTest.java

```diff
@@ -0,0 +1,215 @@
+
+        // *** selec Menu QR Code
+        WebElement qCCodeMenu = wait.until(ExpectedConditions.visibilityOfElementLocated(MobileBy.AccessibilityId(testMenuItemQRCode)));
+        qCCodeMenu.click();
+
+        // inject the image - provide the transformed image to the device with this command
+        String qrCodeImage = encoder("src/test/java/com/realdevice/image_injection/images/qr-code.png");
+        androidDriver.executeScript("sauce:inject-image=" + qrCodeImage);
+
+        // Verify that the browser is running
+        isAndroidBrowserOpened();
+
+        // This is not need only for the video
+        try {
+            Thread.sleep(2000);
+        } catch (InterruptedException ex) {
+            Thread.currentThread().interrupt();
+        }
+
+        androidDriver.quit();
+    }
+
+    @Test
+    public void imageInjectIOS() throws InterruptedException {
+        capabilities.setCapability("deviceName", "iPhone 8*");
+        capabilities.setCapability("platformName", "iOS");
+        capabilities.setCapability("automationName", "XCUITEST");
+
+//      You can use  storage:filename=" +appName if you uploaded your app to Saucd Storage
+//        capabilities.setCapability("app", "storage:filename=" +appName);
+        capabilities.setCapability("app", "https://github.com/saucelabs/my-demo-app-rn/releases/download/v1.0.0-build-61-123/iOS-Real-Device-MyRNDemoApp.1.0.0-67.ipa");
+        capabilities.setCapability("autoAcceptAlerts", true);
+
+        iOSDriver = new IOSDriver(url, capabilities);
+
+        WebDriverWait wait = new WebDriverWait(iOSDriver, 10);
+
+        // *** The main test *****
+
+        // Select QR Code Scanner from the menu
+
+        //wait for the product field to be visible and store that element into a variable
+        WebElement menu = wait.until(ExpectedConditions.visibilityOfElementLocated(MobileBy.AccessibilityId(ios_testMenu)));
+        menu.click();
+
+        // *** selec Menu QR Code
+        WebElement qCCodeMenu = wait.until(ExpectedConditions.visibilityOfElementLocated(MobileBy.AccessibilityId(testMenuItemQRCode)));
+        qCCodeMenu.click();
+
+        // inject the image - provide the transformed image to the device with this command
+        String qrCodeImage = encoder("src/test/java/com/realdevice/image_injection/images/qr-code.png");
+        iOSDriver.executeScript("sauce:inject-image=" + qrCodeImage);
+
+        // Verify that the browser is running
+        isIosApplicationRunning("com.apple.mobilesafari");
+
+        // This is not need only for the video
+        try {
+            Thread.sleep(2000);
+        } catch (InterruptedException ex) {
+            Thread.currentThread().interrupt();
+        }
+
+        iOSDriver.quit();
+    }
+
+
+    // Android
+    public boolean isAndroidBrowserOpened() throws InterruptedException {
+        int wait = 8;
+        int counter = 0;
+
+        do {
+            if (!androidDriver.currentActivity().contains(".MainActivity")){
```

Comment:
@eyaly i'm curious why we need this if we have the `capabilities.setCapability("appWaitActivity", "com.saucelabs.mydemoapp.rn.MainActivity");` set above?

Reply:
This is about the business flow of using the image injection in the demo app. We are in the Native App, we inject the image and we should be switch to the android browser.  so this is a negative check (there is '!' in the beginning, we check that we are NOT in the native app, meaning we switch to the chrome or android browser.